mmv1/templates/terraform/encoders/spanner_instance_config.go.erb

```diff
@@ -0,0 +1,13 @@
+project, err := tpgresource.GetProject(d, meta.(*transport_tpg.Config))
+if err != nil {
+return nil, err
+}
+newObj := make(map[string]interface{})
+newObj["instanceConfig"] = obj
+if obj["name"] == nil {
+  return nil, fmt.Errorf("Error setting instance config name: %s", err)
+}
+newObj["instanceConfigId"] = obj["name"]
+newObj["instanceConfig"][name]= fmt.Sprintf("projects/%s/instances/%s", project, obj["name"])
```

Comment:
It looks like there are some compilation errors currently: https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/8717614383/job/23913172567

```
Error: google-beta/services/spanner/resource_spanner_instance_config.go:576:26: invalid operation: cannot index newObj["instanceConfig"] (map index expression of type interface{})
Error: google-beta/services/spanner/resource_spanner_instance_config.go:576:27: undefined: name
```

This seems to be coming from resource_spanner_instance_config.go:576 - link in the diff: https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10341-old..auto-pr-10341#diff-d278e2518b1b6cc3dfd3b86a4cdec41af738964bcefcd068b0f576039558308bR576

It looks like the problem is that while `obj` is a `map[string]interface{}`, once it's copied into newObj it is implicitly converted to an `interface{}`. I think if you modify `obj` first, it should work as expected.

```suggestion
if obj["name"] == nil {
  return nil, fmt.Errorf("Error setting instance config name: %s", err)
}
newObj["instanceConfigId"] = obj["name"]
obj["name"]= fmt.Sprintf("projects/%s/instances/%s", project, obj["name"])
newObj["instanceConfig"] = obj
```